src/providers/useRegistryAddress.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  CHILD_GAUGE_FACTORY_NAME,
+  useMasterRegistry,
+} from "../hooks/useContract"
+import { UseQueryResult, useQuery } from "@tanstack/react-query"
+import { ChainId } from "../constants"
+import { parseBytes32String } from "ethers/lib/utils"
+import { useActiveWeb3React } from "../hooks"
+
+export type RegistryAddress = Partial<Record<string, string>>
+
+export const useRegistryAddress = (): UseQueryResult<RegistryAddress> => {
+  const { chainId } = useActiveWeb3React()
+  const masterRegistry = useMasterRegistry()
+
+  return useQuery(["registryAddress"], async () => {
```

Comment:
Also reminder to add with a map or enum of react query keys to avoid misspellings